Flickr/Sources/FlickrAppVCs/SearchResultsVC.swift

```diff
@@ -6,17 +6,183 @@
+        imageView.sizeToFit()
+        Nuke.loadImage(with: photos[indexPath.row], into: imageView)
+        cell.addSubview(imageView)
+
+        return [
+            imageView.leadingAnchor.constraint(equalTo: cell.leadingAnchor),
+            imageView.trailingAnchor.constraint(equalTo: cell.trailingAnchor),
+            imageView.topAnchor.constraint(equalTo: cell.topAnchor),
+            imageView.bottomAnchor.constraint(equalTo: cell.bottomAnchor)
+        ]
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        view.backgroundColor = viewBackgroundColor
+        view.backgroundColor = .white
         navigationController?.navigationBar.tintColor = navigationBarTitleColor
+        let constraints = setupCollectionView()
+        NSLayoutConstraint.activate(constraints)
+        fetchPhotos(searchString: searchString ?? "")
+    }
+
+    private func returnSearchUrl(searchString: String) -> URL? {
+        URL(string: "\(baseSearchUrl)/?method=\(apiMethod)&api_key=\(apiKey)&format=\(format)&nojsoncallback=\(noJsonCallback)&text=\(searchString)")
+    }
+
+    private func returnImageURl(image: SinglePhoto) -> URL? {
+        URL(string: "\(imageSearchUrl)/\(image.server)/\(image.id)_\(image.secret).jpg")
+    }
+
+    private func constructIndividualUrls(_ result: Photos) -> [URL] {
+        var individualPhotoUrls: [URL] = []
+        for photo in result.photos.photo {
+            guard let imageUrl = returnImageURl(image: photo) else {
+                return []
+            }
+            individualPhotoUrls.append(imageUrl)
+        }
+        return individualPhotoUrls
+    }
+
+    func fetchPhotos(searchString: String) {
+        var photosArray: [URL] = []
+        guard let url = returnSearchUrl(searchString: searchString) else {
+            return
+        }
+        let task = URLSession.shared.dataTask(with: url, completionHandler: { data, _, error in
+            guard let data = data, error == nil else {
+                return
+            }
+            var result: Photos?
+            do {
+                result = try JSONDecoder().decode(Photos.self, from: data)
+            } catch {
+                return
+            }
+            guard let result = result else {
+                return
+            }
+            photosArray = self.constructIndividualUrls(result)
+            self.photos = photosArray
+            DispatchQueue.main.async {
+                self.collectionView?.reloadData()
+            }
+        })
+        task.resume()
+    }
+
+    func setupCollectionView() -> [NSLayoutConstraint] {
+        let collectionViewLayout = UICollectionViewFlowLayout()
+        collectionViewLayout.minimumInteritemSpacing = minimumInteritemSpacing
+        collectionViewLayout.minimumLineSpacing = minimumLineSpacing
+        collectionViewLayout.sectionInset = sectionInset
+        collectionViewLayout.itemSize = CGSize(width: view.frame.width / widthDivisor, height: cellHeight)
```

Comment:
Please segregate this out into a separate function. This adds too much complication into a single function.